dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/vitals/JankStatsActivityLifecycleListener.kt

```diff
@@ -70,6 +76,27 @@ internal class JankStatsActivityLifecycleListener(
 
     @MainThread
     override fun onActivityStopped(activity: Activity) {
+        val window = activity.window
+        if (!activeActivities.containsKey(window)) {
+            internalLogger.log(
+                InternalLogger.Level.WARN,
+                InternalLogger.Target.MAINTAINER,
+                "Activity stopped but window was not tracked"
+            )
+        }
+        val list = activeActivities[window] ?: mutableListOf()
+        list.removeAll {
+            it.get() == null || it.get() == activity
+        }
+        activeActivities[window] = list
+        if (list.isEmpty()) {
+            internalLogger.log(
+                InternalLogger.Level.DEBUG,
+                InternalLogger.Target.MAINTAINER,
+                "Disabling jankStats for window $window"
+            )
+            activeWindowsListener[window]?.isTrackingEnabled = false
```

Comment:
I'm wondering if we should use `WeakHashMap` for `activeActivities` and `activeWindowsListener`, in case if the window was temporary and will be destroyed once a particular activity is destroyed.

I can see that `PolicyManager.makeNewWindow` can be called from different places, `Dialog` constructor included. So if there is a dialog shown and then destroyed, we will be keeping `Window` instance in the memory forever, preventing it from be GCed.